### ISSUE TITLE
Install doc incomplete

### ISSUE THREAD

**Issue by MrBE4R** [bgo-bioimagerie/platformmanager#228]
Hi,

While installing the current version at Nantes using http://bgo-bioimagerie.github.io/platformmanager/install.html I encoutered some issues.

In the section ```First connection``` : 
Step ```Run php cli/pfm-cli.php install to install/upgrade database and routes``` :
```
PHP Warning:  require(/opt/platformmanager/cli/../vendor/autoload.php): Failed to open stream: No such file or directory in /opt/platformmanager/cli/pfm-cli.php on line 2
PHP Fatal error:  Uncaught Error: Failed opening required '/opt/platformmanager/cli/../vendor/autoload.php' (include_path='.:/usr/share/php') in /opt/platformmanager/cli/pfm-cli.php:2
Stack trace:
#0 {main}
  thrown in /opt/platformmanager/cli/pfm-cli.php on line 2
```
We are missing the step ```composer.phar update``` and composer as a requirement.

While installing the the needed files, I got another error : 
```
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - spipu/html2pdf[v5.2.0, ..., v5.2.1] require php ^5.4 || ^7.0 -> your php version (8.0.8) does not satisfy that requirement.
    - spipu/html2pdf v5.2.2 requires php ^5.6 || ^7.0 -> your php version (8.0.8) does not satisfy that requirement.
    - Root composer.json requires spipu/html2pdf ^5.2 -> satisfiable by spipu/html2pdf[v5.2.0, v5.2.1, v5.2.2].
```
spipu/html2pdf is not yet compatible with php 8.

The upgrade task return a lot of error on the first run
```
pfm.INFO: Create admin user {"admin":""} {"user":"anonymous"}
pfm.ERROR: [sql] error {"sql":"INSERT INTO core_users (login, pwd, name, firstname, email, status_id, source, date_created, apikey) VALUES(?,?,?,?,?,?,?,?,?)","params":["","REDACTED","admin","admin","",2,"local","2021-07-26",""],"error":"SQLSTATE[HY000]: General error: 1364 Field 'date_last_login' doesn't have a default value"} {"user":"anonymous"}
[2021-07-26T11:58:48.567151+02:00] pfm.ERROR: [sql] error {"sql":"CREATE TABLE IF NOT EXISTS `core_users_settings` (
\t\t`user_id` int(11) NOT NULL,
\t\t`setting` varchar(30) NOT NULL DEFAULT '',
\t\t`value` varchar(40) NOT NULL DEFAULT '',
        `id_space` int(11) NOT NULL,
\t\t);","params":null,"error":"SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')' at line 6"} {"user":"anonymous"}
[...]
pfm.ERROR: [sql] error {"sql":"INSERT INTO core_history (id_space, user, message) VALUES (?,?,?)","params":["11",null,"User charbonnier-b joined space"],"error":"SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'user' cannot be null"} {"user":"anonymous"}
pfm.ERROR: [stats] stat error {"message":"Call to a member function createWriteApi() on null"} {"user":"anonymous"}
PHP Warning:  Undefined array key "_user" in /opt/platformmanager/Framework/Events.php on line 54
/// These 2 lines for every users in the database
```
If I run the installer again only the two first error show up again.

When trying to log in after the update I can see in apache's log
```
pfm.ERROR: [sql] error {"sql":"INSERT INTO core_j_spaces_user (id_user, id_space, status) VALUES (?,?,?)","params":["940","11","1"],"error":"SQLSTATE[HY000]: General error: 1364 Field 'convention_url' doesn't have a default value"} {"user":"anonymous"}
```

After altering the database to provide a default value, the ldap login still does not work, I'm redirected to the login page without any message.

**Comment by osallou** [comment]
Indeed from source some steps are missing such as composer install step but you can get all setup required (with php required extensions) in Docker file  (docker-platformmanager repo).

We install everything via docker, manual install steps doc need to be updated.

Php8 is needed, and as in dockerfile we force some deps install which still refer to php7 but works with php8

**Comment by osallou** [comment]
Following errors: 

pfm.ERROR: [sql] error {"sql":"INSERT INTO core_history (id_space, user, message) VALUES (?,?,?)","params":["11",null,"User charbonnier-b joined space"],"error":"SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'user' cannot be null"} {"user":"anonymous"} pfm.ERROR: [stats] stat error {"message":"Call to a member function createWriteApi() on null"} {"user":"anonymous"} PHP Warning: Undefined array key "_user

Can be safely ignored at upgrade (we need to fix this but has no impact)

**Comment by osallou** [comment]
Error 
[2021-07-26T11:58:48.567151+02:00] pfm.ERROR: [sql] error {"sql":"CREATE TABLE IF NOT EXISTS `core_users_settings` ( \t\t`user_id` int(11) NOT NULL,

Has already been fixed, may be not in master branch yet, but can also be ignored

**Comment by osallou** [comment]
However error:
pfm.INFO: Create admin user {"admin":""} {"..

Shows that super admin user is not set in env variables or conf ini file

admin_user =
admin_password = 
admin_email = 
admin_apikey =

We tested with docker via env variables, but should work with settings un conf.ini file

**Comment by osallou** [comment]
For error
pfm.ERROR: [sql] error {"sql":"INSERT INTO core_j_spaces_user (id_user...

We did not face this issue at all, but it seems you start with an existing db.

In which order did you set things?

You should:

Create database with mysql 8
Import your dump in database
Run the pfm cli install command which will update old data

If needed to run cli install again, you must delete in db the records in table pfm_db (which records in which schema version we are, removing it will force a full upgrade again)

**Comment by osallou** [comment]
The error pfm.ERROR: [sql] error {"sql":"INSERT INTO core_j_spaces_user (id_user....

 must be linked to the fact you log with ldap (we don t).
Indeed conventionurl column has no default value, so ldap code should set some value here (not an issue for standard login)

For debug on login, you can set env variables

DEBUG=1

 or conf ini settings:

debug=1

This will set log level to debug

**Comment by osallou** [comment]
Why is your mysql version?

At admin creation time,  we have no error about default values (should just be null for non defined ones), could be a pb with mysql version

**Comment by osallou** [comment]
Some columns in our model do not have a not null constraint, so no error if not defined.

Maybe your dump has a different schema with some not null constraints not present in our schema, which could explain those errors we did not face.

In this case, we need to find a way to remove those constraints, but for that we need to find differences on columns after upgrade....

**Comment by osallou** [comment]
After test and analysis, errors related to non default values occurs with mariadb, not with mysql (tested on v8).

Though table creation sql does not specify a non null on column, it seems to expect this with mariadb.

Doc specifies usage of mysql.

Some tests shoumd be done to try to work with mariadb too

**Comment by osallou** [comment]
Sql mode may need to be set to empty to disable strict mode with mariadb (what is done in our case with mysql)

**Comment by osallou** [comment]
Seems indeed that strict mode should be disabled, a default not null is added at column creation

**Comment by osallou** [comment]
@MrBE4R you have to set sqlmode to empty or at least remove strict mode (see docker compose for mysql)

I will fix code to remove not null constraints on columns where not necessary (which should be fine in strict mode).

Previous code was setting constraint everywhere.

**Comment by gaetanHerve** [comment]
solved with commit 7ccf9b4b3117c7d4c6c0397d115f8d9f859e4a0b
